sdk/language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -78,45 +78,52 @@ optionsParserInfo = info (optionsParser <**> helper)
     <> progDesc "Generate TypeScript bindings from a DAR"
     )
 
+readPackageNameAndVersion :: String -> (PackageName, PackageVersion)
+readPackageNameAndVersion nameAndVersion = (PackageName name, PackageVersion version)
```

Comment:
The package name that is embedded in the dar is actually a combination of two pieces from the original `daml.yaml` - the name and the version, connected with a hyphen. We need to distinguish these pieces when we read the dar.